tests/kafkatest/services/kafka/kafka.py

```diff
@@ -226,10 +229,20 @@ def prop_file(self, node):
                                  security_config=self.security_config, num_nodes=self.num_nodes)
         return prop_file
 
+    def file_exists(self, node, filename_regex):
+        try:
+            node.account.ssh("test -s %s" % filename_regex, allow_fail=False)
+            return True
+        except RemoteCommandError:
+            return False
+
     def start_cmd(self, node):
         cmd = "export JMX_PORT=%d; " % self.jmx_port
-        cmd += "export KAFKA_LOG4J_OPTS=\"-Dlog4j.configuration=file:%s\"; " % self.LOG4J_CONFIG
+        if self.do_logging:
+            cmd += "export KAFKA_LOG4J_OPTS=\"-Dlog4j.configuration=file:%s\"; " % self.LOG4J_CONFIG
```

Comment:
Can you explain the intent of this? Usually if we want to turn off logging, we would just set the log level to OFF or something like that. Not sure that's actually a good idea for this test since we may need to debug failures.

Reply:
Hmm, good point. I found that, using the standard template, we ran out of memory because of logging itself. I think this was because it tried to dump out the full fetch response. I will check to see if we can dial down the log level to avoid that.

Reply:
In the worst case, we should be able to set log levels to OFF, though this might require some template adjustment. However, as mentioned offline, we should probably figure out what is causing the OOM in logging and fix it.